hack/bin/flaky_tests.py

```diff
@@ -0,0 +1,187 @@
+    else:
+        return s2
+
+def is_flake(item):
+    b = item['build']
+    return ((b['pipeline_name'] == 'mls' and b['job_name'] == 'test') \
+            or (b['pipeline_name'] == 'staging' and b['job_name'] == 'test') \
+            or (b['pipeline_name'] == 'prod' and b['job_name'] == 'test'))
+
+
+def add_flake(flake_set, test_name):
+    for k in flake_set:
+        if tests_match(k, test_name):
+            k_ = longer(k, test_name)
+            if k_ != k:
+                flake_set.remove(k)
+                flake_set.add(k_)
+            return
+    flake_set.add(test_name)
+
+def discover_flakes(data):
+    flake_set = set()
+    for item in data:
+        if is_flake(item):
+            add_flake(flake_set, item['test_name'])
+    return flake_set
+
+def search_matching_flake(test_names, test_name):
+    for flake in test_names:
+        if tests_match(flake, test_name):
+            return flake
+
+def associate_fails(test_names, data):
+    d = {k: [] for k in test_names}
+    for item in data:
+        flake = search_matching_flake(test_names, item['test_name'])
+        if flake:
+            d[flake].append(item)
+    return [{'test_name': k, 'fails': v} for k, v in  d.items()]
+
+def associate_comments(flakes, comments):
+    for flake in flakes:
+        flake['comments'] = comments.get(flake['test_name'])
+
+def sort_flakes(flakes):
+    flakes.sort(key=lambda f: len(f['fails']), reverse=True)
+    for flake in flakes:
+        flake['fails'].sort(key=lambda f: f['build']['end_time'], reverse=True)
+
+def humanize_days(n):
+    if n < 7:
+        if n == 0:
+            return 'today'
+        elif n == 1:
+            return 'yesterday'
+        else:
+            return f'{n} days ago'
+    else:
+        weeks = n // 7
+        if weeks < 4:
+            return f'{weeks} weeks ago'
+        else:
+            return f'{weeks // 4} months ago'
+
+def human_format_date(dt, today):
+    days = (today - dt).days
+    return f'· {format_date(dt)} ({humanize_days(days)})'
+
+def create_url(build):
+    return CONCOURSE_BASEURL + f"/pipelines/{build['pipeline_name']}/jobs/{build['job_name']}/builds/{build['name']}"
+
+def pretty_flake(flake, today):
+    lines = []
+    lines.append(Colors.YELLOW + f"❄ \"{flake['test_name']}\"" + Colors.RESET)
+    comments = flake['comments']
+    if comments:
+        for l in comments.splitlines():
+            lines.append('    ' + l)
+    else:
+        lines.append('    (discovered flake, no notes in flaky-tests.yaml)')
+    lines.append('')
+    for fail in flake['fails']:
+        b = fail['build']
+        end_time = datetime.datetime.fromtimestamp(b['end_time'])
+        s = human_format_date(end_time, today)
+        if (today - end_time) < datetime.timedelta(days=CONCOURSE_LOG_RETENTION_DAYS):
+            url = create_url(b)
+            s = s + ' ' + url
+        lines.append('  ' + s)
+
+    return "\n".join(lines) + '\n'
+
+def pager(s):
+    pipe = os.popen('less -S', 'w')
```

Comment:
```suggestion
    pipe = os.popen('less -RS', 'w')
```
Adding `-R` here makes `less` render escape sequences correctly.